Does not build in IBM i PASE environment

### Bug Report
In the IBM i PASE (it's based on IBM AIX) environment with PHP + extensions installed from `http://repos.zend.com/ibmiphp` and further extensions installed from `http://repo.qseco.fr`, the command `pecl install mongodb` downloads but the build fails as shown in the attached document.
[pecl_mongodb_fail.txt](https://github.com/mongodb/mongo-php-driver/files/5403692/pecl_mongodb_fail.txt)


### Environment
IBM i 7.4 PASE
```
$ uname -a
OS400 OPENSRC 4 7 00210002307W Os
```

```$ ./build_driver.sh 
Submodule 'src/libmongoc' (https://github.com/mongodb/mongo-c-driver.git) registered for path 'src/libmongoc'
Submodule 'src/libmongocrypt' (https://github.com/mongodb/libmongocrypt.git) registered for path 'src/libmongocrypt'
Cloning into '/home/JWOEHR/work/PHP/mongo-php-driver/src/libmongoc'...
Cloning into '/home/JWOEHR/work/PHP/mongo-php-driver/src/libmongocrypt'...
Submodule path 'src/libmongoc': checked out '60267532b6feddad7ab93097d8917fa1dd7195df'
Submodule path 'src/libmongocrypt': checked out '052f7fc610f0cea83a2adf3dd263a5ff04833371'
configure: WARNING: You will need re2c 0.13.4 or later if you want to regenerate PHP parsers.
configure: WARNING: *** Please add powerpc-ibm-os400 to configure.ac checks!
/home/JWOEHR/work/PHP/mongo-php-driver/src/libmongoc/src/libbson/src/bson/bson-oid.c:35:23: warning: 'gHexCharPairs' defined but not used [-Wunused-const-variable=]
 static const uint16_t gHexCharPairs[] = {
                       ^~~~~~~~~~~~~
/home/JWOEHR/work/PHP/mongo-php-driver/src/libmongoc/src/libmongoc/src/mongoc/mongoc-socket.c: In function 'mongoc_socket_accept_ex':
/home/JWOEHR/work/PHP/mongo-php-driver/src/libmongoc/src/libmongoc/src/mongoc/mongoc-socket.c:731:15: error: 'struct sockaddr_storage' has no member named 'ss_family'; did you mean '__ss_family'?
       if (addr.ss_family == AF_INET) {
               ^
/home/JWOEHR/work/PHP/mongo-php-driver/src/libmongoc/src/libmongoc/src/mongoc/mongoc-socket.c: In function '_mongoc_socket_try_sendv':
/home/JWOEHR/work/PHP/mongo-php-driver/src/libmongoc/src/libmongoc/src/mongoc/mongoc-socket.c:1295:10: warning: implicit declaration of function 'sendmsg' [-Wimplicit-function-declaration]
    ret = sendmsg (sock->sd,
          ^~~~~~~
make: *** [Makefile:491: src/libmongoc/src/libmongoc/src/mongoc/mongoc-socket.lo] Error 1
```

```
$ cat ../build_driver.sh 
BREP=..build_report_` date '+%Y%m%dT%H:%M:%S'`.txt
export OBJECT_MODE=64
phpize >${BREP}&& \
CFLAGS="-D_BSD=44 -pthread" CPPFLAGS="-D_BSD=44 -pthread" LDFLAGS="-Wl,-bbigtoc -pthread" ./configure --enable-mongodb-developer-flags >>${BREP}&& \
make clean >>${BREP}&& make all >>${BREP}&& make install >>${BREP}
```
Same error.
```
/home/JWOEHR/work/PHP/mongo-php-driver/src/libmongoc/src/libmongoc/src/mongoc/mongoc-socket.c: In function 'mongoc_socket_accept_ex':
/home/JWOEHR/work/PHP/mongo-php-driver/src/libmongoc/src/libmongoc/src/mongoc/mongoc-socket.c:731:15: error: 'struct sockaddr_storage' has no member named '__ss_family'; did you mean 'ss_family'?
       if (addr.ss_family == AF_INET) {
               ^
make: *** [Makefile:491: src/libmongoc/src/libmongoc/src/mongoc/mongoc-socket.lo] Error 1
```

--- a/scripts/autotools/libmongoc/FindDependencies.m4
+++ b/scripts/autotools/libmongoc/FindDependencies.m4
@@ -26,8 +26,8 @@ AC_CHECK_TYPE([socklen_t],
               [#include <sys/socket.h>])
 
 AC_CHECK_MEMBER([struct sockaddr_storage.ss_family],
-                [AC_SUBST(MONGOC_HAVE_SS_FAMILY, 0)],
                 [AC_SUBST(MONGOC_HAVE_SS_FAMILY, 1)],
+                [AC_SUBST(MONGOC_HAVE_SS_FAMILY, 0)],
                 [#include <sys/socket.h>])
 
 # Check for pthreads. libmongoc's original FindDependencies.m4 script did not

